### PR TITLE
Preserve ordering of dependencies

### PR DESCRIPTION
**Pull request by hadley**
(If they're ordered)

The logic is complicated, but the tests hit every line, so I'm reasonable certain it's correct.

### CODE REVIEW

File: R/deps.R [hunk 1]
```diff
@@ -6,15 +6,28 @@ idesc_set_dep <- function(self, private, package, type, version) {
...
-      )
+    row <- data.frame(
+      stringsAsFactors = FALSE,
+      type = type, package = package, version = version
     )
+    others <- deps$package[deps$type == type]
+    sorted <- !is.unsorted(others) && length(others) > 0
+
+    if (sorted) {
+      # find first row it should come after
```
Comment (gaborcsardi) [inline R/deps.R]: Can't we just add it to the bottom, and then sort the df, if it was already sorted?

Comment (hadley) [reply R/deps.R]: But you have to consider the sorting within a section (i.e. the imports might be sorted but the suggests is not). I tried a few approaches and this seemed the most robust to me (despite being rather complicated)

Comment (gaborcsardi) [reply R/deps.R]: Right. I suspected that it is not so simple.

**Comment by gaborcsardi** [comment]
Thanks!

**Pull request merged**